app/forms/bulk_submission_form.rb

```diff
@@ -78,4 +92,96 @@ def file_size(file)
     return 0 if file.nil?
     File.size(file&.tempfile)
   end
+
+  def csv
+    @csv ||= CSV.parse(File.read(uploaded_file), headers: true)
```

Comment:
just had a thought - we should squish whitespace. the way i have done it on my branch is

```ruby
  def csv_table
    @csv_table ||= CSV.parse(
                      content,
                      headers: :first_row,
                      header_converters: lambda { |f| f.strip },
                      converters: lambda { |f| f ? f.strip : nil }
                    )
  end
```

which should work here too. should probably have a test for that too 😬

Reply:
Have added your suggestion, and included a spec to test that validation passes for a csv with whitespace in headers (and content)